.github/workflows/continuous-integration.yml

```diff
@@ -356,34 +384,51 @@ jobs:
           DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
           DOCKERHUB_PASSWORD: ${{ secrets.DOCKERHUB_PASSWORD }}
           DOCKER_IMAGE_NAME: ${{ secrets.DOCKER_IMAGE_NAME }}
+          # Output from build_base job
+          VERSION: ${{ needs.build_base.outputs.VERSION }}
+          BRANCH: ${{ needs.build_base.outputs.BRANCH }}
         shell: bash
         run: |
           set -eu -o pipefail
 
           for var in DOCKERHUB_USERNAME DOCKERHUB_PASSWORD DOCKER_IMAGE_NAME; do
-              if [[ ! -v $var || -z ${!var} ]]; then
-                  echo "Required variable $var is not set." >&2
-                  exit 1
-              fi
+            if [[ ! -v $var || -z ${!var} ]]; then
+              echo "Required variable $var is not set." >&2
+              exit 1
+            fi
           done
 
-          version=$(cat artifacts-docker/version.txt)
-          image_name=$DOCKER_IMAGE_NAME
+          suffix=""
+          ci_run=""
 
-          if [[ $GITHUB_REF == refs/heads/* ]]; then
-              image_name+=-staging
-          fi
+          # Add suffix if trying or merging
+          if [[ $BRANCH =~ ^(trying|staging)$ ]]; then
+            suffix="-$BRANCH"
 
-          docker tag coop.rchain/rnode:latest "$image_name:$version"
-          docker tag coop.rchain/rnode:latest "$image_name:latest"
+            # Add CI run number if trying
+            if [[ $BRANCH =~ ^trying$ ]]; then
+              ci_run="-ci-$GITHUB_RUN_NUMBER"
+            fi
+          fi
 
           builtin echo "$DOCKERHUB_PASSWORD" \
               | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
 
           set -x
-          docker push "$image_name:$version"
-          docker push "$image_name:latest"
 
+          image_name="$DOCKER_IMAGE_NAME$suffix"
+          img_version="$image_name:$VERSION$ci_run"
```

Comment:
Just curious. Do we really need the ci run number?

Reply:
I've added it because _trying_ can be done multiple times and with force push to the PR related commit can be difficult to track. Run number can make it easier to track which image belongs to which _try_, and also to be even more distinguishable from _staging_ image.